### PR TITLE
Add MongoDB support for deadletter refactor

### LINKED ISSUE #60: Update Mongo store apis
There were a number of changes to the way stores are used (e.g. for deadletter we added a number of apis) and the Mongo store needs to be updated .  Currently Crawler in a Box is broken as it uses Mongo.

### PR DESCRIPTION
**Pull request by willbar**
Resolves #60

There's some oddities attempting to mimic the behavior of the Azure Blob store. I want to come back and resolve these but it's more to have CIAB running successfully.

### CODE REVIEW

File: providers/storage/mongodocstore.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ class MongoDocStore {
     if (cached) {
       return Q(cached.document);
     }
-    return this.db.collection(type).findOne({ '_metadata.url': url }).then(value => {
+    return this.db.collection(type).findOne({ '$or': [{ '_metadata.url': url }, { '_metadata.links.self.href': url }] }).then(value => {
```
Comment (jeffmcaffer) [inline providers/storage/mongodocstore.js]: why are you looking at the links?  that href should be the urn.

If you do want to do the or here and test for both, you likely need to add the same code in etag()

File: providers/storage/mongodocstore.js [hunk 2]
```diff
@@ -59,19 +59,30 @@ class MongoDocStore {
...
-    return Q(true);
+    return this.db.collection(type).deleteOne({ $or: [{ '_metadata.url': url }, { '_metadata.links.self.href': url }] }).then(result => {
+      return result;
+    });
   }
 
-  count(pattern) {
-    // TODO implement
-    return Q(0);
+  count(type) {
```
Comment (jeffmcaffer) [inline providers/storage/mongodocstore.js]: Gene just implemented the table store that I think uses "pattern".  Would be good to update that wording as well.

File: providers/storage/mongodocstore.js [hunk 3]
```diff
@@ -59,19 +59,30 @@ class MongoDocStore {
...
+        metadata.urn = metadata.links.self.href;
+        return metadata;
+      })
+    });
   }
 
   delete(type, url) {
-    // TODO implement
-    return Q(true);
+    return this.db.collection(type).deleteOne({ $or: [{ '_metadata.url': url }, { '_metadata.links.self.href': url }] }).then(result => {
```
Comment (jeffmcaffer) [inline providers/storage/mongodocstore.js]: same comment here about the or and use of href.  if this is prevalent, can we factor out the query object and share it across all the functions?

File: providers/storage/mongodocstore.js (hunk 2 above)
Comment (geneh) [inline providers/storage/mongodocstore.js]: The file name is tableMappingStore.

**Pull request merged**